ndk-build/src/apk.rs

```diff
@@ -124,11 +126,13 @@ impl<'a> UnalignedApk<'a> {
         let abi = target.android_abi();
         let lib_path = Path::new("lib").join(abi).join(path.file_name().unwrap());
         let out = self.config.build_dir.join(&lib_path);
-        std::fs::create_dir_all(out.parent().unwrap())?;
+        fs::create_dir_all(out.parent().unwrap())?;
 
         match self.config.strip {
             StripConfig::Default => {
-                std::fs::copy(path, out)?;
+                fs::copy(path, out.clone())?;
+                #[cfg(target_family = "unix")]
+                fs::set_permissions(out, fs::Permissions::from_mode(0o644))?;
```

Comment:
This should at the very least have a comment documenting this workaround for Nix :wink: